rewrite-core/src/main/java/org/openrewrite/internal/RemoveAfter.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.internal;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+/**
+ * An annotation that indicates that the element should be removed after a certain point.
+ * Intended to be used with a matching recipe to clear out elements on a schedule.
+ */
+@Retention(RetentionPolicy.SOURCE)
+public @interface RemoveAfter {
+    /**
+     * Expects a date without a time-zone in the ISO-8601 calendar system, such as 2007-12-03.
+     * @return The date after which this element should be removed.
+     */
+    String date();
```

Comment:
I think an optional `description` attribute would make sense to have.

Also, what do you think about some kind of boolean `processable` attribute (what to name this thing?) defaulting to `false`, which would allow a recipe to both remove the annotation and the thing it annotates? I think it sounds interesting, but I wonder how often it is actually actionable. Don't you typically also have to adjust other code too?

Reply:
I'd briefly looked at adding a description, but then realized this is intended mostly to be used along side `@Deprecated` to signal that usage downstream should move to an alternative implementation, so i figured there would not be a need to  duplicate that message.

As to the `processable` I'm not sure that's necessary? Or how were you intending to use that when true vs false?

I'd mostly intended this annotation when we say add a field to a class, and want to add a backwards compatible constructor. It could then make sense to both deprecate and mark for future removal, to give folks some time to adapt.

Do you think we'd need anything more here? Or can we either merge or close if not deemed immediately usable?

Reply:
That could work as well; then I'll close this PR and we can look at structured comments when we find time to build a recipe.